ssl/statem/statem_clnt.c

```diff
@@ -1136,6 +1136,21 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
                session_id_len);
     }
 
+    if (!ssl_verify_client_session_version(s)) {
```

Comment:
This is what BoringSSL does, along with every other client I've tested. If the version doesn't match, it's a fatal error. Switching to a fresh session isn't an option anyway. `s->hit` is still 1, so we'll try to resume an empty master secret. (And setting `s->hit` back to 0 will cause the handshake to break. The server will continue the handshake as if it's a resumption handshake.)